OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/backend/impl/InAppBackendService.kt

```diff
@@ -209,4 +202,84 @@ internal class InAppBackendService(
     ) {
         Logging.error("Encountered a $statusCode error while attempting in-app message $requestType request: $response")
     }
+
+    private suspend fun attemptFetchWithRetries(
+        baseUrl: String,
+        rywToken: String,
+        sessionDurationProvider: () -> Long,
+    ): List<InAppMessage>? {
+        var attempts = 1
+        var delayTime = 1 // Start with a 1-second delay for exponential backoff
+        var retryLimit = DEFAULT_RETRY_LIMIT
+
+        // We use RYW_TOKEN_NOT_PROVIDED to designate the backend not returning a RYW token -- due
+        // to a potential regression or feature flag
+        if (rywToken != ConsistencyManager.RYW_TOKEN_NOT_PROVIDED) {
+            while (attempts <= retryLimit + 1) {
+                val retryCount = if (attempts > 1) attempts - 1 else null
+                val values =
+                    OptionalHeaders(
+                        rywToken = rywToken,
+                        sessionDuration = sessionDurationProvider(),
+                        retryCount = retryCount,
+                    )
+                val response = _httpClient.get(baseUrl, values)
+
+                if (response.isSuccess) {
+                    val jsonResponse = response.payload?.let { JSONObject(it) }
+                    return jsonResponse?.let { hydrateInAppMessages(it) }
+                } else if (response.statusCode == 425) { // 425 Too Early
+                    if (response.retryLimit != null) {
+                        retryLimit = response.retryLimit!!
+                    }
+                    val retryAfter = response.retryAfterSeconds ?: DEFAULT_RETRY_AFTER_SECONDS
+                    delay(retryAfter * 1_000L)
+                } else if (response.statusCode == 429) { // 429 Too Many Requests
+                    val retryAfter = response.retryAfterSeconds ?: delayTime
+
+                    delay(retryAfter * 1_000L)
+                    delayTime *= 2 // Exponential backoff
```

Comment:
I'm not sure if we want to implement the exponential backoff in the SDK. I feel like if we need it then we should add that in turbine. I think it would be more predictable if the SDK just adheres to the Retry-After header instead of overriding it with it's own exponential backoff

Reply:
Done.